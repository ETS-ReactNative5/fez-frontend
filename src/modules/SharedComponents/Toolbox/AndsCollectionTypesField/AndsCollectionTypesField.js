import React from 'react';
import { GenericSelectField } from 'modules/SharedComponents/GenericSelectField';
import { ANDS_COLLECTION_TYPE_COLLECTION, ANDS_COLLECTION_TYPE_DATASET } from 'config/general';

export default function AndsCollectionTypesField(fieldProps) {
    return (
        <GenericSelectField
            hideLabel={false}
            locale={{ label: fieldProps.label }}
            value={fieldProps.input.value}
            onChange={!!fieldProps.input && fieldProps.input.onChange}
            itemsList={[
                {
                    value: ANDS_COLLECTION_TYPE_COLLECTION,
                    text: 'Collection',
                },
                {
                    value: ANDS_COLLECTION_TYPE_DATASET,
                    text: 'Dataset',
                },
            ]}
            errorText={(!!fieldProps.meta && fieldProps.meta.error) || ''}
            error={(!!fieldProps.meta && !!fieldProps.meta.error) || false}
            {...fieldProps}
        />
    );
}
