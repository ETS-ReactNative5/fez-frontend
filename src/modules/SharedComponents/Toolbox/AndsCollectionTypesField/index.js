export { default as AndsCollectionTypesField } from './AndsCollectionTypesField';
