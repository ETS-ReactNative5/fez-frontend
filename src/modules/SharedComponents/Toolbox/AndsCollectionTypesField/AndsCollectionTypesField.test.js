import AndsCollectionTypesField from './AndsCollectionTypesField';

function setup(testProps = {}) {
    const props = {
        label: 'Collection type',
        input: {
            value: 453615,
            onChange: jest.fn(),
        },
        errorText: '',
        error: false,
        ...testProps,
    };

    return getElement(AndsCollectionTypesField, props);
}

describe('AndsCollectionTypesField component', () => {
    it('should render default view', () => {
        const wrapper = setup();

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
