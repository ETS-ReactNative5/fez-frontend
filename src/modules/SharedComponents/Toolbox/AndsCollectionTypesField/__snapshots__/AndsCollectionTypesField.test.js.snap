// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AndsCollectionTypesField component should render default view 1`] = `
<GenericSelectField
  error={false}
  errorText=""
  genericSelectFieldId="rek-ands-collection-type"
  hideLabel={false}
  input={
    Object {
      "onChange": [MockFunction],
      "value": 453615,
    }
  }
  itemsList={
    Array [
      Object {
        "text": "Collection",
        "value": 453615,
      },
      Object {
        "text": "Dataset",
        "value": 453616,
      },
    ]
  }
  label="Collection type"
  locale={
    Object {
      "label": "Collection type",
    }
  }
  onChange={[MockFunction]}
  value={453615}
/>
`;

exports[`AndsCollectionTypesField component should render error and error message 1`] = `
<GenericSelectField
  error={false}
  errorText=""
  genericSelectFieldId="rek-ands-collection-type"
  hideLabel={false}
  input={
    Object {
      "onChange": [MockFunction],
      "value": 453615,
    }
  }
  itemsList={
    Array [
      Object {
        "text": "Collection",
        "value": 453615,
      },
      Object {
        "text": "Dataset",
        "value": 453616,
      },
    ]
  }
  label="Collection type"
  locale={
    Object {
      "label": "Collection type",
    }
  }
  meta={
    Object {
      "error": "This field is required",
    }
  }
  onChange={[MockFunction]}
  value={453615}
/>
`;
